stellargraph/layer/deep_graph_infomax.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import GCN, GAT, APPNP, PPNP
+from ..core.experimental import experimental
+
+from tensorflow.keras.layers import Input, Lambda, Layer, GlobalAveragePooling1D
+import tensorflow as tf
+from tensorflow.keras import backend as K
+import warnings
+
+__all__ = ["DeepGraphInfoMax", "BilinearTransform"]
+
+
+class BilinearTransform(Layer):
+    """
+    This Layer computes the Discriminator function for Deep Graph Infomax (https://arxiv.org/pdf/1809.10341.pdf).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def build(self, input_shapes):
+
+        self.kernel = self.add_weight(
+            shape=(input_shapes[0][-1], input_shapes[0][-1]),
```

Comment:
As a minor code style thing, this could be rephrased to be a bit more self-explanatory:

```python
feature_size = input_shapes[0][-1]
self.kernel = self.add_weight(
    shape=(feature_size, feature_size)
```

Or, alternatively, more general:

```python
first_size = input_shapes[0][-1]
second_size = input_shapes[1][-1]
self.kernel = self.add_weight(
    shape=(feature_size, second_size)
```

and then the `summary` argument doesn't _have_ to match the `features` one (cf. the discussion about the bilinear name above).